src/views/Dashboard/components/Widgets/PortfolioWidget/PortfolioWidget.tsx

```diff
@@ -0,0 +1,168 @@
+  //
+  // Need to pass in quantity of 1 if it's an NFT, otherwise need to request a quantity.
+  // Using asset name for now but might be flawed if a user creates an asset with a name of "nftleda", or
+  // if Leda changes the way they tag their NFTs
+  const handleSendAsset = useCallback(async (assetId: string, assetName: string) => {
+    if (assetId === undefined || assetName === undefined) {
+      console.error("inadequate details provided to handleSendAsset, please try again");
+      return;
+    }
+
+    setAssetSendId(assetId);
+
+    if (assetName === LedaNFTName) {
+      console.log(`Asset is ${assetName} with ID: ${assetId}, forcing a send qty of 1`);
+      setAssetSendQty("1");
+    }
+
+    console.log("collecting additional tx details before seed collection...");
+    setCollectTxDetails(true);
+  }, []);
+
+  const handleNext = useCallback(async () => {
+    if (sendAsset === undefined || assetToAddress === undefined || assetSendQty === undefined || assetSendId === undefined) {
+      // enqueue a snackbar here
+      return;
+    }
+
+    console.log("proceeding to next dialog...");
+
+    setCollectTxDetails(false);
+    const result = await sendAsset(assetToAddress, assetSendQty, assetSendId);
+
+    console.log("sendWidget sendJUP result:", result);
+  }, [assetSendId, assetSendQty, assetToAddress, sendAsset]);
+
+  const handleClose = useCallback(() => {
+    setCollectTxDetails(false);
+  }, []);
+
+  const portfolioRows: Array<ITableRow> | undefined = useMemo(() => {
+    if (heldAssets === undefined || !Array.isArray(heldAssets)) {
+      return undefined;
+    }
+
+    return heldAssets.map((asset) => {
+      return {
+        assetId: asset.asset,
+        assetName: asset.name,
+        assetBalance: asset.quantityQNT,
+        assetDescription: asset.description,
+        actions: (
+          <Stack direction={"row"} spacing={2} justifyContent="center">
+            <Button variant="green" onClick={() => handleSendAsset(asset.asset, asset.name)}>
+              Send
+            </Button>
+            <Button variant="green" onClick={() => handleCopyAssetId(asset.asset)}>
+              Copy Asset ID
+            </Button>
+          </Stack>
+        ),
+      };
+    });
+  }, [handleCopyAssetId, handleSendAsset, heldAssets]);
+
+  return (
+    <>
+      <JUPTable
+        title={"My Portfolio"}
+        path={"/portfolio"}
+        headCells={headCells}
+        rows={portfolioRows}
+        defaultSortOrder="asc"
+        keyProp={"assetId"}
+      ></JUPTable>
+      {collectTxDetails ? (
+        <JUPDialog isOpen={collectTxDetails} closeFn={handleClose}>
+          <Stack sx={{ alignItems: "center" }} spacing={2}>
+            <JUPAddressInput placeholder='Enter "To" Address' fetchAddressFn={(address) => fetchToAddress(address)}></JUPAddressInput>
+            <JUPQuantityInput placeholder="Enter Quantity" fetchQuantityFn={(quantity) => fetchAssetQuantity(quantity)}></JUPQuantityInput>
```

Comment:
I didn't like the duplication here either, hadn't considered passing in the validation function though. 

I also don't quite know how it will evolve over time but I *think* it should work as one component. 

https://github.com/Up-North-Coding/jovian/issues/62